src/Control/Email/Email.php

```diff
@@ -276,14 +276,30 @@ public function setSwiftMessage($swiftMessage)
         $dateTime = new DateTime();
         $dateTime->setTimestamp(DBDatetime::now()->getTimestamp());
         $swiftMessage->setDate($dateTime);
-        if (!$swiftMessage->getFrom() && ($defaultFrom = $this->config()->get('admin_email'))) {
-            $swiftMessage->setFrom($defaultFrom);
+        if (!$swiftMessage->getFrom()) {
+            $swiftMessage->setFrom($this->getDefaultFrom());
         }
         $this->swiftMessage = $swiftMessage;
 
         return $this;
     }
 
+    /**
+     * @return string
+     */
+    private function getDefaultFrom(): string
+    {
+        $defaultFrom = $this->config()->get('admin_email');
+        if (!$defaultFrom) {
+            $host = parse_url(Director::host(), PHP_URL_HOST);
+            // $host may be blank during unit tests
```

Comment:
Is it only unit tests or all CLI calls?

Reply:
CLI, though this was failing in a CI context where there wasn't a websever.  CLI in a normal context, Director::host() should work. If it doesn't, then I don't really know what else we can do here.

Reply:
I'd say check whether this is run in tests and then allow for example.com, otherwise log an error or possibly even throw an exception.

Reply:
I agree with logging if not throwing an error outside of tests - and even in the test perhaps it would make sense to set `PHP_URL_HOST` in the test instead, so the test is actually checking that this works as expected.

Reply:
I've updated this to update config in EmailTest to set the host, and to throw an error if host is not set

The fact the host was blank was actually a misuse of `parse_url(Director::host(), PHP_URL_HOST)` instead of just `Director::host()` - `parse_url('www.mysite.com', PHP_URL_HOST)` will return null, though `parse_url('https://www.mysite.com', PHP_URL_HOST)` will return `www.mysite.com`.

In a CI context, Direct::host() should return `localhost`